File: 89. Gray Code/89. Gray Code.go
func grayCode(n int) []int {
    size := 1 << n  // 2^n
    result := make([]int, size)
    
    for i := 0; i < size; i++ {
        result[i] = i ^ (i >> 1)
    }
    
    return result
}

File: 89. Gray Code/89. Gray Code.rb
# @param {Integer} n
# @return {Integer[]}
def gray_code(n)
    size = 1 << n  # 2^n
    result = Array.new(size)
    
    (0...size).each do |i|
        result[i] = i ^ (i >> 1)
    end
    
    result
end